src/Fmt_odoc.ml

```diff
@@ -63,23 +65,22 @@ let fmt_verbatim_block s =
   let content =
     (* Literal newline to avoid indentation *)
     if force_break then wrap "\n" "@\n" (str s)
-    else
-      fits_breaks " " "\n"
-      $ str (String.strip s)
-      $ fits_breaks " " ~hint:(0, 0) ""
+    else fits_breaks " " "\n" $ str s $ fits_breaks " " ~hint:(0, 0) ""
   in
   hvbox 0 (wrap "{v" "v}" content)
 
-let fmt_code_block s =
-  let fmt_line ~first ~last:_ l =
-    let l = String.rstrip l in
-    if first then str l
-    else if String.length l = 0 then str "\n"
-    else fmt "@," $ str l
-  in
-  let lines = String.split_lines s in
-  let box = match lines with _ :: _ :: _ -> vbox 0 | _ -> hvbox 0 in
-  box (wrap "{[@;<1 2>" "@ ]}" (vbox 0 (list_fl lines fmt_line)))
+let fmt_code_block conf s =
+  try hvbox 0 (wrap "{[@;<1 2>" "@ ]}" (conf.fmt_code conf.conf s))
```

Comment:
That `fmt_code` is expected to raise should be documented in the `mli`. Or a more explicit type should be used, eg. `(Fmt.t, unit) result`.